lib/SGN/Model/solGS/solGS.pm

```diff
@@ -1171,14 +1155,26 @@ sub _create_genotype_row {
         my $genotype =  $genotype_hash->{$marker};
 	$genotype =  $genotype_hash->{$marker};
 
-        $geno_values .= $genotype =~ /\d+/g ? $round->round($genotype) : $genotype;       
+        $geno_values .= $self->round_allele_dosage_values($genotype);       
         $geno_values .= "\t" unless $marker eq $markers[-1];
     }
 
     return $geno_values;
 
 }
 
+sub round_allele_dosage_values {
+    my ($self, $geno_values) = @_;
+
+    my $round = Math::Round::Var->new(0);
```

Comment:
Indeed slow... on another on going branch I will look into doing the rounding in R (where there should be faster methods). 

In the interest of other applications/users that would use the genotype data, alternatively loading rounded dosage values to the database would be a solution.